File: app/src/main/java/kz/shymkent/relaxhouse/addClientFragment/AddClientDialogFragment.java
package kz.shymkent.relaxhouse.addClientFragment;

import android.app.AlertDialog;
import android.app.DatePickerDialog;
import android.app.Dialog;
import android.app.TimePickerDialog;
import android.os.Build;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.DatePicker;
import android.widget.TimePicker;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.RequiresApi;
import androidx.databinding.DataBindingUtil;
import androidx.fragment.app.DialogFragment;
import androidx.lifecycle.ViewModelProvider;

import java.util.Calendar;

import kz.shymkent.relaxhouse.R;
import kz.shymkent.relaxhouse.databinding.FragmentAddClientBinding;
import kz.shymkent.relaxhouse.mainActivity.MainActivityViewModel;
import kz.shymkent.relaxhouse.models.Client;

public class AddClientDialogFragment extends DialogFragment {
    AddClientDialogFragmentViewModel viewModel;
    FragmentAddClientBinding fragmentAddClientBinding;
    public AlertDialog alertDialog;
    Calendar calendar = Calendar.getInstance();
    Client client = new Client();
    @RequiresApi(api = Build.VERSION_CODES.N)
    @NonNull
    @Override
    public Dialog onCreateDialog(@Nullable Bundle savedInstanceState) {
        viewModel = new ViewModelProvider(this).get(AddClientDialogFragmentViewModel.class);
        final AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(getActivity());
        LayoutInflater inflater = getActivity().getLayoutInflater();
        fragmentAddClientBinding = DataBindingUtil.inflate(inflater,R.layout.fragment_add_client,null,false);
        View view = fragmentAddClientBinding.getRoot();
        alertDialogBuilder.setView(view);
        alertDialogBuilder.setTitle("Новый клиент");
        alertDialog = alertDialogBuilder.create();
        fragmentAddClientBinding.setClient(client);
        fragmentAddClientBinding.setAddClientDialogFragment(this);
        return alertDialog;
    }

    @RequiresApi(api = Build.VERSION_CODES.N)
    public void showCheckInDatePickerDialog( ){
        DatePickerDialog datePickerDialog = new DatePickerDialog(getContext());
        DatePickerDialog.OnDateSetListener listener = new DatePickerDialog.OnDateSetListener() {
            @Override
            public void onDateSet(DatePicker view1, int year, int month, int dayOfMonth) {
                client.checkInDate = dayOfMonth + "." + (month + 1) + "." + year;
                showCheckInTimePickerDialog();
            }
        };
        datePickerDialog.setOnDateSetListener(listener);
        datePickerDialog.show();
    }
    @RequiresApi(api = Build.VERSION_CODES.N)
    public void showCheckOutDatePickerDialog(){
        DatePickerDialog datePickerDialog = new DatePickerDialog(getContext());
        datePickerDialog.setOnDateSetListener(new DatePickerDialog.OnDateSetListener() {
            @Override
            public void onDateSet(DatePicker view1, int year, int month, int dayOfMonth) {
                client.checkOutDate = dayOfMonth + "." + (month + 1) + "." + year;
                showCheckOutTimePickerDialog();
            }
        });
        datePickerDialog.show();
    }
    private void showCheckInTimePickerDialog(){
        TimePickerDialog timePickerDialog = new TimePickerDialog(getContext(), new TimePickerDialog.OnTimeSetListener() {
            @Override
            public void onTimeSet(TimePicker view, int hourOfDay, int minute) {
                client.checkInTime = (hourOfDay + ":" + String.format("%02d", minute));
                fragmentAddClientBinding.invalidateAll();
            }
        },calendar.get(Calendar.HOUR_OF_DAY),calendar.get(Calendar.MINUTE),true);
        timePickerDialog.show();
    }
    private void showCheckOutTimePickerDialog(){
        TimePickerDialog timePickerDialog = new TimePickerDialog(getContext(), new TimePickerDialog.OnTimeSetListener() {
            @Override
            public void onTimeSet(TimePicker view, int hourOfDay, int minute) {
                client.checkOutTime = (hourOfDay + ":" + String.format("%02d", minute));
                fragmentAddClientBinding.invalidateAll();
            }
        },calendar.get(Calendar.HOUR_OF_DAY),calendar.get(Calendar.MINUTE),true);
        timePickerDialog.show();
    }
    public void addClient(Client client){
        if(!client.getCheckInDate().isEmpty() && !client.getCheckInTime().isEmpty() && ! client.getCheckOutDate().isEmpty() && !client.getCheckOutTime().isEmpty()
                && !client.getQuantity().isEmpty() && !client.getAvans().isEmpty() && !client.getDebt().isEmpty() && !client.getPhoneNumber().isEmpty()){
            viewModel.addClientToFireBaseDB(client);
            Toast.makeText(getContext(), "Новый клиент добавлен", Toast.LENGTH_SHORT).show();
            alertDialog.dismiss();
        }else{
            Toast.makeText(getContext(), "Заполните пустые поля", Toast.LENGTH_SHORT).show();
        }

    }
    public void cancel(){
        alertDialog.dismiss();
    }
}

File: app/src/main/java/kz/shymkent/relaxhouse/addClientFragment/AddClientDialogFragment.kt
package kz.shymkent.relaxhouse.addClientFragment

import android.app.AlertDialog
import android.app.DatePickerDialog
import android.app.TimePickerDialog
import android.os.Build
import android.os.Bundle
import android.widget.Toast
import androidx.annotation.RequiresApi
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.DialogFragment
import androidx.lifecycle.ViewModelProvider
import kz.shymkent.relaxhouse.R
import kz.shymkent.relaxhouse.databinding.FragmentAddClientBinding
import kz.shymkent.relaxhouse.mainActivity.MainActivityViewModel
import kz.shymkent.relaxhouse.models.Client
import java.util.*

class AddClientDialogFragment : DialogFragment() {
    val client = Client()
    private lateinit var alertDialog : AlertDialog
    private lateinit var binding : FragmentAddClientBinding
    private val calendar = Calendar.getInstance()
    private lateinit var viewModel : MainActivityViewModel


    override fun onCreateDialog(savedInstanceState: Bundle?): AlertDialog {
        viewModel = ViewModelProvider(requireActivity())[MainActivityViewModel::class.java]
        val alertDialogBuilder = AlertDialog.Builder(activity)
        binding = DataBindingUtil.inflate(layoutInflater, R.layout.fragment_add_client, null, false)
        alertDialogBuilder.setView(binding.root)
        alertDialogBuilder.setTitle(getString(R.string.new_client))
        alertDialog = alertDialogBuilder.create()
        binding.client = client
        binding.addClientDialogFragment = this
        return alertDialog
    }

    @RequiresApi(Build.VERSION_CODES.N)
    fun showCheckInDatePickerDialog(){
        val datePickerDialog = DatePickerDialog(requireContext())
        datePickerDialog.setOnDateSetListener { _, year, month, dayOfMonth ->
            client.checkInDate = "$dayOfMonth." + (month + 1) + ".$year"
            showCheckInTimePickerDialog()
        }
        datePickerDialog.show()
    }

    @RequiresApi(Build.VERSION_CODES.N)
    fun showCheckOutDatePickerDialog(){
        val datePickerDialog = DatePickerDialog(requireContext())
        datePickerDialog.setOnDateSetListener { _, year, month, dayOfMonth ->
            client.checkOutDate = "$dayOfMonth." + (month + 1) + ".$year"
            showCheckOutTimePickerDialog()
        }
        datePickerDialog.show()
    }

    private fun showCheckInTimePickerDialog(){
        val timePickerDialog = TimePickerDialog(context,
            { _, hourOfDay, minute ->
                client.checkInTime = "$hourOfDay:" + String.format("%02d", minute)
                binding.invalidateAll()
            },calendar[Calendar.HOUR_OF_DAY], calendar[Calendar.MINUTE], true)
        timePickerDialog.show()
    }

    private fun showCheckOutTimePickerDialog(){
        val timePickerDialog = TimePickerDialog(context,
            { _, hourOfDay, minute ->
                client.checkOutTime = "$hourOfDay:" + String.format("%02d", minute)
                binding.invalidateAll()
            },calendar[Calendar.HOUR_OF_DAY], calendar[Calendar.MINUTE], true)
        timePickerDialog.show()
    }
    fun addClient(client : Client){
        if(client.checkInDate.isNotEmpty() && client.checkInTime.isNotEmpty() && client.checkOutDate.isNotEmpty() && client.checkOutTime.isNotEmpty()
            && client.quantity.isNotEmpty() && client.avans.isNotEmpty() && client.debt.isNotEmpty() && client.phoneNumber.isNotEmpty()
        ) {
            viewModel.addClientToFirebase(client)
            Toast.makeText(context, getString(R.string.new_client_added), Toast.LENGTH_SHORT).show()
            alertDialog.dismiss()
        }else{
            Toast.makeText(context, getString(R.string.fill_empty_fields), Toast.LENGTH_SHORT).show()
        }
    }

    fun cancel(){
        alertDialog.dismiss()
    }
}